File: lab-report-03/counting-sort.c
//Fatima
//Batch: E-76th
//Roll:19

//counting sort ascending or descending order

#include<stdio.h>
    int main(){
        int a[50],n,i,j,temp;


        printf("Enter how many numbers you want to insert:");
        scanf("%d",&n);
        printf("\nEnter the numbers :");
        for(i=0; i<n;i++){
            scanf("%d",&a[i]);
        }

         for(i=0; i<n; i++){
            for(j=i+1; j<n; j++){
                if(a[i]>a[j]){
                    temp=a[i];
                    a[i] = a[j];
                    a[j] = temp;
                }
            }
        }

        printf("Sorted list in ascending order:");
        for(i=0; i<n; i++){
        printf("%d",a[i]);
        }

        printf("\nSorted list in descending order:");
        for(i=n-1; i>=0;i--){
        printf("%d", a[i]);
        }

    return 0;
}

File: lab-report-03/counting-sort.js
var A = [10,1,6,8,12,20];
var C = [];
var B = [];

//Given that 
console.log('1');
console.log("Enter how many numbers you want to insert: 6");
console.log("Enter the numbers: 10 1 6 8 12 20");

//Menu section
console.log(".............Menu...............");
console.log("Press 1 for ascending order sorting");
console.log("Press 2 for descending order sorting");

//Choice section
var choice = prompt("Enter your choice:");
console.log("Enter your choice:"+ choice);

//counting sort for ascending order sorting
if(choice == 1){
    for(var i = 0; i<=20; i++){
        C[i] = 0;
    }
    for(var j = 0; j<6; j++){
        C[A[j]] = C[A[j]] + 1;
    }
    for(var i = 1; i<=20; i++){
        C[i] = C[i] + C[i-1];
    }
    for(var j = 5; j>= 0; j--){
        B[C[A[j]]] = A[j];
        C[A[j]] = C[A[j]] - 1;
    }

    //output section
    console.log("Sorted list in ascending order:");
        for(var i = 1; i<=6; i++){
            console.log(B[i]);
        }
    }